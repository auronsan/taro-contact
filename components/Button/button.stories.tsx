import React from 'react';
import { Button } from './index';

export default {
  title: 'components/Button',
  component: Button,
};

export const withOnClick = () => <Button onClick={() => {}}>Click Me</Button>;

export const withChildren = () => <Button>Click Me</Button>;

export const withCustomClassName = () => <Button className="custom-class">Click Me</Button>;

export const withRedColor = () => <Button c="red">Click Me</Button>;

export const withOutlineVariant = () => <Button v="outline">Click Me</Button>;

export const withFlexStyle = () => <Button flex={2}>Click Me</Button>;
