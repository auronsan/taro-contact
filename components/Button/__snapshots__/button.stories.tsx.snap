// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button withChildren smoke-test 1`] = `
<main class="roboto-normal">
  <button type="button"
          class="button_root__s7zfG button_button-primary__cWRWo button_button-variant-filled__Dz0lC"
  >
    Click Me
  </button>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Button withCustomClassName smoke-test 1`] = `
<main class="roboto-normal">
  <button type="button"
          class="button_root__s7zfG button_button-primary__cWRWo button_button-variant-filled__Dz0lC custom-class"
  >
    Click Me
  </button>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Button withFlexStyle smoke-test 1`] = `
<main class="roboto-normal">
  <button type="button"
          class="button_root__s7zfG button_button-primary__cWRWo button_button-variant-filled__Dz0lC"
          style="flex: 2 1 0%;"
  >
    Click Me
  </button>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Button withOnClick smoke-test 1`] = `
<main class="roboto-normal">
  <button type="button"
          class="button_root__s7zfG button_button-primary__cWRWo button_button-variant-filled__Dz0lC"
  >
    Click Me
  </button>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Button withOutlineVariant smoke-test 1`] = `
<main class="roboto-normal">
  <button type="button"
          class="button_root__s7zfG button_button-primary__cWRWo button_button-variant-outline__hdUZT"
  >
    Click Me
  </button>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;

exports[`components/Button withRedColor smoke-test 1`] = `
<main class="roboto-normal">
  <button type="button"
          class="button_root__s7zfG button_button-red__lET9U button_button-variant-filled__Dz0lC"
  >
    Click Me
  </button>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;
